x10/perpetual/transfer_object.py

```diff
@@ -0,0 +1,77 @@
+import math
+from datetime import timedelta
+from decimal import Decimal
+from typing import List
+
+from x10.perpetual.accounts import AccountModel, StarkPerpetualAccount
+from x10.perpetual.markets import MarketModel
+from x10.perpetual.transfers import PerpetualTransferModel, StarkTransferSettlement
+from x10.utils.date import utc_now
+from x10.utils.model import SettlementSignatureModel
+from x10.utils.starkex import generate_nonce, get_transfer_msg
+
+SECONDS_IN_HOUR = 60 * 60
+ASSET_ID_FEE = 0
+MAX_AMOUNT_FEE = 0
+
+
+def find_account_by_id(accounts: List[AccountModel], account_id: int):
+    return next((account for account in accounts if account.account_id == account_id), None)
+
+
+def calc_expiration_timestamp():
+    expire_time = utc_now() + timedelta(days=7)
+    expire_time_with_buffer = expire_time + timedelta(days=7)
+    expire_time_with_buffer_as_hours = math.ceil(expire_time_with_buffer.timestamp() / SECONDS_IN_HOUR)
+
+    return expire_time_with_buffer_as_hours
+
+
+def create_transfer_object(
+    from_account_id: int,
+    to_account_id: int,
+    amount: Decimal,
+    transferred_asset: str,
+    stark_account: StarkPerpetualAccount,
+    accounts: List[AccountModel],
+    market: MarketModel,
+):
+    from_account = find_account_by_id(accounts, from_account_id)
+    to_account = find_account_by_id(accounts, to_account_id)
+
+    expiration_timestamp = calc_expiration_timestamp()
+    stark_amount = (amount * market.collateral_asset.settlement_resolution).to_integral_exact()
```

Comment:
Not sure about this one as `to_stark_amount` has a different logic, but the current one is aligned with FE